src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -1092,3 +1118,43 @@ function getSessionId(): string {
 
     return sessionId;
 }
+
+async function createClientPipeTransport(
+    pipeName: string,
+    encoding: RAL.MessageBufferEncoding = 'utf-8'
+): Promise<PipeTransport> {
+    const connected = new Promise<[MessageReader, MessageWriter]>((resolve) => {
+        const server: net.Server = net.createServer((socket: net.Socket) => {
+            // Sever has connected
+            server.close();
```

Comment:
If the server never connects (say because something else is broken and we're crashing on startup) are we going to be leaking pipes?